### ISSUE TITLE
clang-tidy fails due to warnings in non-user code

### ISSUE THREAD

**Issue by MarkKoz** [pocc/pre-commit-hooks#36]
If I include a standard header e.g. `stdio.h`, clang-tidy will analyse it and fail because the header's content doesn't comply with the checks:

> 2816 warnings generated.
Suppressed 2816 warnings (2816 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.

In the context of pre-commit, this is a problem because it causes the process to exit with code 1. Therefore, the check fails even though these warnings are being "suppressed". clang-tidy clearly knows these warnings are not from user code, yet I cannot find a way to make it stop exiting with code 1 (`--quiet` doesn't help). I'm a novice with clang-tidy, so I may have misconfigured something.

With this in mind, how is it even practical to use clang-tidy at all with pre-commit? How do you use it?

**Comment by MarkKoz** [comment]
Actually, it only seems to exit with code 1 through pre-commit. If I invoke `clang-tidy` directly, it shows the warnings were suppressed but exits with code 0.

```
clang-tidy...............................................................Failed
- hook id: clang-tidy
- exit code: 1

2816 warnings generated.
Suppressed 2816 warnings (2816 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

```yaml
repos:
    - repo: https://github.com/pocc/pre-commit-hooks
      rev: v1.3.3
      hooks:
          - id: clang-format
            args: [-i, --style=file]
          - id: clang-tidy
            args: [--fix, -p=cmake-build-debug]
```

**Comment by pocc** [comment]
What version of llvm are you using? What OS are you on? The test ok file uses <stdio.h> and the test passes.  I'll take a look at it.

**Comment by pocc** [comment]
Can you provide your `src/main.c` as well (or a link to the repo that has it)? I'm not able to replicate this.

**Comment by pocc** [comment]
Adding some tests for double-dash arguments `--` `--std=c++20`, I got a test that failed (try using -- args with your local hook). I'm guessing the default language target on your system is higher than mine? 

Most recent commit is 661e54b - can you set rev to that and see if you still have issues?

**Comment by MarkKoz** [comment]
Sorry for a late response. I did get together a minimal reproducible example, but I tried that new ref, and it indeed fixes the issue. I was going to point out that clang-tidy was outputting to stderr and the hook always failed if it saw anything in stderr, but you caught onto that already.

For posterity, I suppose I'll attach the example anyway since I already made it. It can be reproduced through Docker.

[test.zip](https://github.com/pocc/pre-commit-hooks/files/7117985/test.zip)

> I'm guessing the default language target on your system is higher than mine?

I don't know; I specify it through CMake. I thought clang-tidy somehow determines the language target by reading `compile_commands.json`.
